### DISCUSSION TITLE
Obstructions disappear when simulation starts

### DISCUSSION THREAD

**Discussion by nsarifds** [firemodels/fds#12800] [Q&A] [answered]
Hello everyone, I have posted some pictures to describe my problem. At image 1 you can see all of my cable targets setup in Pyrosim. Image 2 is the beginning of the simulation, all my obstructions have disappeared. Image 3 is a slice of the temperature, there the obstructions are visible through the different temperature gradient. At this temperature these obstructions should have ignited. Image 4 is the HRR in the simulation volume, it is equal to the fire's HRR, therefore there was no ignition of the cables. Any ideas what is happening? This is the FDS version just before the release of 6.9.0.

![1](https://github.com/firemodels/fds/assets/161325698/8c839475-dbf7-4ac7-bf97-511a2b7fc63f)
![2](https://github.com/firemodels/fds/assets/161325698/278992d3-b9e8-46a6-bb3f-e7258cef150d)
![3](https://github.com/firemodels/fds/assets/161325698/9f756a06-10b2-4d18-a936-32a6eeed8f4c)
![4](https://github.com/firemodels/fds/assets/161325698/04b26899-94a9-468e-8135-846d788951ec)

**Comment by rmcdermo** [comment]
Please post an input file.

**Reply by nsarifds** [reply]
[test.txt](https://github.com/firemodels/fds/files/14978155/test.txt)
Here you go! The fire vent is 2.53m2 and 1000 HRRPUA, however only half of it ignites, I'm assuming the floor is not on one level, this is unimportant for now. Also you can split the meshes at Z=13.4 to get 32 meshes instead of 16. I have failed with both. You will also notice 0-thickness obstructions and pressure zones. First step is to ignite the cables and get numerical instability error, then I would try to use ULMAT and the pressure zone volume limit.

**Reply by rmcdermo** [reply]
Do you think it could have anything to do with COLOR='INVISIBLE' on the OBST line?

**Reply by nsarifds** [reply]
Its only the obstructions named Kabeltrasse and KKabeltrasse that disappear, the invisible obstructions are other building elements that I have "hidden" in order to interact easier with the model in Pyrosim. I have tried running my model with ULMAT and I get error -4 on the first time step. I also tried with limiting the Pressure Zone volume to 0.2m3 and I also got an error on the first time step.

**Reply by rmcdermo** [reply]
OK, well, unfortunately, my system is down right now, so it will take me some time to dig into this.  It can't hurt to try the latest release in the mean time.  6.9.1 was posted a few days ago.

**Reply by nsarifds** [reply]
I think I have also failed with 6.9.1 but let me try again to be sure

**Reply by nsarifds** [reply]
Yeah, they still disappear even with 6.9.1

**Reply by nsarifds** [reply]
[test_2.fds.txt](https://github.com/firemodels/fds/files/14979166/test_2.fds.txt)

Here's the input with only the relevant obstructions, still disappearing when run with 6.9.1

**Reply by nsarifds** [reply]
![image](https://github.com/firemodels/fds/assets/161325698/7c8d0c46-baee-4827-bc59-6681171c2832)

Fire plume goes right through them

**Comment by mcgratta** [comment, marked as answer]
This appears to be a bug. Try it without the `BURN_AWAY` feature.

**Reply by nsarifds** [reply]
That's it, the obstructions make it past the 1st second! However I'm not really sure how to realistically monitor the progress of the fire without the burnaway feature. I think the expected behavior now is for the HRR to rise as more trays ignite, but there will never be any indication that fuel has run out for the HRR to drop. Any chance of seeing a fix?

**Comment by mcgratta** [comment]
Yes, I have expanded the functionality of thin obstructions, but there have been some bugs that have developed. I'll fix.

**Comment by mcgratta** [comment]
OK, here is what I think the problem is. When you specify `BULK_DENSITY`, FDS determines the combustible mass of the obstruction to be the `BULK_DENSITY` multiplied by the volume of the obstruction (`OBST`). In your case, the volume of your obstructions as determined by the `XB` values on the `OBST` line is zero. Thus, the obstructions have no mass and "burn away" on the first time step. If you want to maintain the `BULK_DENSITY` of 120 kg/m3, then make your obstructions 1 cm thick. FDS will still make them appear to be flat, but it will at least know that you want the obstruction to have some combustible mass. Let me know if that works. I will add logic to stop the simulation if a `BULK_DENSITY` is applied to an `OBST` with no volume.

**Reply by nsarifds** [reply]
![image](https://github.com/firemodels/fds/assets/161325698/e940f3c2-fe17-41c6-bd16-583334005929)

Did it for 6 OBST lines instead of the whole 1800, I guess they're the only ones that made it through to the sim

**Comment by IFAB-AS** [comment]
If you use BURN_AWAY and BULK_DENSITY in Pyrosim and set the thickness of the OBST to 1 cm using a coarser mesh (e.g. 5 cm), the OBST will still be blocked to a zero thickness OBST by Pyrosim. This means, FDS will still interpret this as an OBST with 0 m³ with no mass.
The issue is currently work in progress at Thunderhead, as I noticed the same problem several weeks ago. FDS is working correctly (uses values from the OBST line), but Pyrosim should use THICKEN for these lines to avoid this error or not use the BULK_DENSITY for these entries but the THICKNESS and standard density.
Additionally, if other blocks are resized due to meshing the BULK_DENSITY for each OBST should be calculated automatically to give the same total mass for the OBST.

**Reply by nsarifds** [reply]
So Thunderhead is working on automatically thickening these obstructions when you import them? I tried running my code with THICKEN=.TRUE. on every OBST line, but they disappeared again.

**Reply by IFAB-AS** [reply]
I did check in my notes: If you use THICKEN=.TRUE., the OBST line in the FDS file still has a volume of 0 and for this reason the blocks will be removed shortly after starting the simulation. At the beginning at t=0 it should be visible.

**Comment by IFAB-AS** [comment]
The error needs to be fixed in Pyrosim before the FDS lines are exported (= export them as thickened blocks where necessary + recalculation of the BULK_DENSITY) or the calculation in FDS needs to be changed to resolve these cases. Thunderhead is aware of the issue as far as I know their replies to my issue in Pyrosim

**Reply by nsarifds** [reply]
Thank you very much for this information, that would be really helpful when working with imported geometries.

**Comment by mcgratta** [comment]
Our (FDS developers) intent is that the user should be able to specify the exact dimensions and then FDS will internally "snap" them to the grid. The new 3D heat transfer algorithm allows lateral heat transfer in thin plates, in which case the user specifies exactly what the obstruction dimensions are. If Pyrosim does the "snapping", it should not change the obstruction dimensions to zero.
